src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, useState } from "react";
+import { ActionButton, Input, Modal } from "@canonical/react-components";
+import { useFormik } from "formik";
+import Loader from "components/Loader";
+import { fetchIdentityDetails } from "api/auth-identities";
+import { base64EncodeTokenDetails } from "util/helpers";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [isLoadingToken, setIsLoadingToken] = useState(false);
+  const [token, setToken] = useState<string | null>(null);
+
+  const handleClose = () => {
+    onClose();
+    window.location.reload();
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      clientName: "",
+    },
+    onSubmit: (values) => {
+      setIsLoadingToken(true);
+
+      fetchIdentityDetails(values.clientName)
+        .then((tokenPreResponse) => {
+          const tempToken = base64EncodeTokenDetails(tokenPreResponse);
+
+          setIsLoadingToken(false);
+          setToken(tempToken);
+        })
+        .catch((err) => {
+          setIsLoadingToken(false);
+          console.error("Error fetching token:", err);
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={onClose}
+      className="create-tls-identity"
+      title="Generate Trust Token"
+      buttonRow={
+        <>
+          {token ? (
+            <ActionButton
+              appearance="positive"
+              className="u-no-margin--bottom"
+              onClick={() => handleClose()}
+            >
+              Refresh
+            </ActionButton>
+          ) : (
+            <ActionButton
+              appearance="positive"
+              className="u-no-margin--bottom"
+              onClick={() => void formik.submitForm()}
+              disabled={
+                formik.values.clientName.length === 0 ||
+                isLoadingToken ||
+                token != null
+              }
+            >
+              Generate Token
+            </ActionButton>
+          )}
+        </>
+      }
+    >
+      {isLoadingToken && <Loader text="Loading token..." />}
+      {!isLoadingToken && !token && (
```

Comment:
We should disble the input instead of hiding it when making the API call, this could reduce layout shift in the modal. The loading state can be indicated by the action button instead of the dedicated loader

Reply:
Hmm. Won't this approach also result in layout shift because at one point the tokens are there and at the next, they aren't?
In my recent commit, I took out the loader state entirely, but I have still left the input box and token box there and moved the loading feedback to the action button. Let me know what you think?

Reply:
the layout shift would be during the loading state for the modal. If we don't show the loader and just disable the input, then during the loading state, the modal will not have its size adjusted

Reply:
so what you've done now seems good :+1: